src/goe/offload/operation/ddl_file.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2024 The GOE Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import os
+from typing import TYPE_CHECKING
+
+from goe import __version__ as package_version
+from goe.exceptions import OffloadOptionError
+from goe.filesystem.goe_dfs import get_scheme_from_location_uri
+from goe.filesystem.goe_dfs_factory import get_dfs_from_options
+from goe.offload import offload_constants
+from goe.util.misc_functions import standard_file_name
+
+if TYPE_CHECKING:
+    from goe.config.orchestration_config import OrchestrationConfig
+    from goe.goe import OffloadOperation
+    from goe.offload.offload_messages import OffloadMessages
+
+
+DDL_FILE_HEADER = "Table DDL generated by GOE"
+DDL_FILE_HEADER_TEMPLATE = f"""-- {DDL_FILE_HEADER}
```

Comment:
Are the `--` comments supported by all currently-supported backends?

Reply:
Yes, that is a valid single line comment on BigQuery, Snowflake, SQL Server and Impala/Hive.
Do you think the comment is unnecessary? I can take it away if desired.

Reply:
I think the comments are useful. Thanks.